### ISSUE TITLE
Metrics after closing the SparContext

### ISSUE THREAD

**Issue by jfgosselin** [groupon/spark-metrics#26]
If I stop() my SparkSession (and in the same process my SparkContext) and create a new one within the same JVM, there's no way to re-initialized UserMetricsSystem ?

**Comment by jpfourny** [comment]
Nope. But you can do some clever, but **awful**, hacks to reset private data structures in Scala using reflection API. I had to do this for testing purposes. 

Example:
[UserMetricsSystemHelper.java.txt](https://github.com/groupon/spark-metrics/files/1718563/UserMetricsSystemHelper.java.txt)
